### PR TITLE
Note for that EventManager is refreshed for every test method.

### PR DESCRIPTION
**Pull request by makallio85**

### CODE REVIEW

File: en/development/testing.rst [hunk 1]
```diff
@@ -208,6 +208,10 @@ expectations and assertions your tests make. Additionally because tests are
 code, they are easy to re-run whenever you make a change. This helps prevent
 the creation of new bugs.
 
+.. note::
+
+EventManager is refreshed for each test method. Meaning, that when running multiple tests at once, you will loose your event listeners that were registered in config/bootstrap.php as the bootstrap is only executed once.
```
Comment (antograssiot) [inline en/development/testing.rst]: Could you please indent this liNE and split it to 80 chars max ?

Comment (lorenzo) [inline en/development/testing.rst]: `loose` should be change by `lose`

**Comment by lorenzo** [comment]
Should we make another recommendation instead? Like having an events.php file that is loaded in bootstrap and for every test. /cc @markstory

**Comment by markstory** [comment]
@lorenzo So similar to how routesbare loaded?

File: en/development/testing.rst (hunk 1 above)
Comment (makallio85) [inline en/development/testing.rst]: Yep, fixed. Sorry, haven't contributed to docs that much to able to remember guidelines :/

**Comment by makallio85** [comment]
This is actually quite annoying because:
- If I register event listener in bootstrap.php, it is overridden in TestCase::setUp()
- If I register it in AppController, events are not available in my custom ExceptionRenderer or command line interface
- If I register events in events.php I have to do it manually in every test. Of course I could build own testing classes and extend them, but I think there are enough inheritance already there.

So this would be nice have solved somehow. Would it be possible, that bootstrap.php, bootsrap_cli.php and TestCase.php would look for events.php from project and plugins config folders?

**Comment by lorenzo** [comment]
@markstory yes, similar to that

**Comment by markstory** [comment]
@lorenzo Sounds like something we can do in 3.2 or 3.3.

**Comment by dereuromark** [comment]
What shall we do with this one?

File: en/development/testing.rst [hunk 2]
```diff
@@ -208,6 +208,12 @@ expectations and assertions your tests make. Additionally because tests are
 code, they are easy to re-run whenever you make a change. This helps prevent
 the creation of new bugs.
 
+.. note::
+
+    EventManager is refreshed for each test method. Meaning, that when running
```
Comment (ravage84) [inline en/development/testing.rst]: "Meaning, " -> "This means "

**Pull request merged**